tests/host_tools/fcmetrics.py

```diff
@@ -276,29 +309,7 @@ def validate_fc_metrics(metrics):
             ]
             vhost_user_devices.append(metrics_name)
 
-    firecracker_metrics_schema = {
-        "type": "object",
-        "properties": {},
-        "required": [],
-    }
-
-    for metrics_name, metrics_fields in firecracker_metrics.items():
-        metrics_schema = {
-            "type": "object",
-            "required": metrics_fields,
-            "properties": {},
-        }
-        for metrics_field in metrics_fields:
-            if (
-                metrics_name in latency_agg_metrics
-                and metrics_field in latency_agg_metrics[metrics_name]
-            ):
-                metrics_type = "object"
-            else:
-                metrics_type = "number"
-            metrics_schema["properties"][metrics_field] = {"type": metrics_type}
-        firecracker_metrics_schema["properties"][metrics_name] = metrics_schema
-        firecracker_metrics_schema["required"].append(metrics_name)
+    firecracker_metrics_schema = create_metrics_schema_objects(firecracker_metrics)
```

Comment:
nit: would it be sensible to add a "test for test" to make sure that it captures missing metrics? Maybe a simple smoke test with a very sparse input structure to check that the validation rejects it.

Reply:
initially I though about adding some negative test cases but, after adding the "additionalProperties:false" flag in the schema, the test started reporting all the missing metrics including the block/net and latency aggregate. So, I think there is no need for negative tests.

Reply:
fair enough, `additionalProperties:false` looks convincing.